compiler/main/driver.cpp

```diff
@@ -1502,14 +1504,26 @@ static void setPrintCppLineno() {
   if (developer && !userSetCppLineno) printCppLineno = false;
 }
 
-static void setSaveCForGpuCodegen() {
+static void setGPUFlags() {
   bool isGpuCodegen = localeUsesGPU();
   bool isSaveCDirEmpty = strlen(saveCDir) == 0;
 
-  if(isGpuCodegen && isSaveCDirEmpty) {
-    int len = snprintf(saveCDir, FILENAME_MAX+1, "%s_gpu_files", executableFilename);
-    if(len < 0 || len >= FILENAME_MAX+1) {
-      USR_FATAL("Unable to produce name of savec directory for GPU code generation.");
+  if(isGpuCodegen) {
+    if (isSaveCDirEmpty) {
+      const char *gpuDir = strlen(executableFilename) != 0 ? 
+            executableFilename:std::to_string(getpid()).c_str();
+
+      int len = snprintf(saveCDir, FILENAME_MAX+1, "%s_gpu_files", gpuDir);
+      if(len < 0 || len >= FILENAME_MAX+1) {
+        USR_FATAL("Unable to produce name of savec directory for GPU code generation.");
+      }
+    }
+
+    if (!fNoChecks) {
+      fNoChecks = true;
+      USR_WARN("The prototype support for GPUs imply --no-checks."
```

Comment:
`The prototype support for GPUs imply --no-checks.` -> `The prototype GPU support implies --no-checks.`